lib/ui/pages/chat_page.dart

```diff
@@ -19,10 +21,22 @@ class _ChatPageState extends State<ChatPage> {
   final TextEditingController _messageController = TextEditingController();
   final List<ChatMessage> _messages = [];
   final ScrollController _scrollController = ScrollController();
-  final String _mockResponse = MockData.mockResponse;
 
+  late final DataRepositoryImpl _repository;
   bool _isLoading = false;
 
+  @override
+  void initState() {
+    super.initState();
+
+    final groq = GroqGenerator(apiKey: AIConfig.groqApiKey);
```

Comment:
read about dependency injection and inject this dependencies via "getIt" dart package